tests/unit/FirebaseObject.spec.js

```diff
@@ -322,6 +322,21 @@ describe('$FirebaseObject', function() {
       expect($scope.test).toEqual({$value: null, $id: obj.$id, $priority: obj.$priority});
     });
 
+    it('should delete $value if set to an object', function () {
```

Comment:
`utils.updateRec` was already deleting `$value` on the syncObject for primitives, so that should be reflected on `$scope` as well.